### ISSUE TITLE
Can't drag WayPoint in Directions

### ISSUE THREAD

**Issue by hsianchen** [mapbox/mapbox-unity-sdk#1313]
* Unity version: 2018.3.1f1
* Mapbox SDK version: 2.0.0
* The platform you're building to: Android

Hi, I'm trying to add Directions to my project.
This is what I do:
I copy the Directions object from TrafficAndDirections scene, then I past it into my project.
It shows the directions on the map but I can't drag the waypoint in the Game scene when I hit the play.  What did I miss?
Here's my Hierarchy:
![Github_H](https://user-images.githubusercontent.com/25290748/54534866-4451a900-49c8-11e9-9119-4605f6325341.jpg)
Game scene:
![Github_Gamescene](https://user-images.githubusercontent.com/25290748/54534892-503d6b00-49c8-11e9-99cf-f9e8f1c6ab3a.jpg)
Waypoint:
![waypoint](https://user-images.githubusercontent.com/25290748/54535341-5122cc80-49c9-11e9-986a-60addee29424.jpg)
pinpoint:
![pinpoint](https://user-images.githubusercontent.com/25290748/54535374-5f70e880-49c9-11e9-8f66-a4701ab813ae.jpg)

**Comment by jordy-isaac** [comment]
You might have to make some changes and customizations to get it to work. Can you describe a bit more about what you've tried and your implementation?

**Comment by brnkhy** [comment]
Hello @hsianchen,
I quickly tried it, created a prefab of that gameobject by dragging it to project, opened another demo scene and added that prefab to that scene. It worked for me as intended (I was able to move waypoints).
- are you getting any errors?
- have you verified if `OnMouseDrag` method in `DragableDirectionWaypoint` is called when you try dragging waypoints?

**Comment by hsianchen** [comment]
@brnkhy There's only one error (but I don't think this is related to my problem):
![error](https://user-images.githubusercontent.com/25290748/54589050-8c6fda80-4a5f-11e9-842b-5997ee2ac04f.jpg)

I use `WorldScaleAR`  and I'm trying to build an AR navigation app. I created a prefab of Directions and dragged it into my scene.
![Scene](https://user-images.githubusercontent.com/25290748/54591192-b24bae00-4a64-11e9-8567-26899ada9eb9.jpg)


To verify whether `OnMouseDrag` method in `DragableDirectionWaypoint` is really working, I added `Debug.Log("drag");` in `OnMouseDrag`. Like this:
![onmouseScript](https://user-images.githubusercontent.com/25290748/54590939-25a0f000-4a64-11e9-9c8a-df9fc5b290f5.jpg)
And the console:
![console_drag](https://user-images.githubusercontent.com/25290748/54590983-36516600-4a64-11e9-939c-5d0b1c7ed580.jpg)

But I still can't drag the waypoint.

**Comment by brnkhy** [comment]
an error can break some unity functions I guess. still it's surprising to see drag event works but pins doesn't move. It can be anything at this point I guess. Maybe they move but too slow? Or maybe something snaps them to their position? It's hard to say from here I'm afraid :(

**Comment by hsianchen** [comment]
@brnkhy  Sorry, I pasted the wrong error. Here's the correct one:
![console](https://user-images.githubusercontent.com/25290748/54657551-dd85da00-4b04-11e9-8e8f-53ac924ddd7d.jpg)

@jordy-isaac  I created a new `WorldScaleAR` scene and dragged the `Direction` prefab to it. But I didn't see the direction and the waypoint in the **map scene** when I pressed the play.
![all](https://user-images.githubusercontent.com/25290748/54658457-78cc7e80-4b08-11e9-9d6b-e2b5fd676cf9.jpg)
